File: WalkudUI/src/main/java/owncloudsms/walkudui/BaseActivity.java
package owncloudsms.walkudui;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.widget.Toast;

import butterknife.ButterKnife;

/**
 * Activity基类
 * Created by Walkud on 16/9/21.
 */
public abstract class BaseActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(getLayoutId());
        ButterKnife.bind(this);
        initData();
        addListener();
    }

    /**
     * 获取布局Id
     *
     * @return
     */
    public abstract int getLayoutId();

    /**
     * 初始化数据
     */
    public abstract void initData();

    /**
     * 添加事件
     */
    public abstract void addListener();

    /**
     * 显示Toast提示
     *
     * @param msg
     */
    public void showToast(String msg) {
        Toast.makeText(this, msg, Toast.LENGTH_SHORT).show();
    }

    protected void toIntent(Class cls) {
        Intent intent = new Intent(this, cls);
        startActivity(intent);
    }
}

File: mystudy/WalkudApp/src/main/kotlin/com.walkud.self/mvp/base/BaseActivity.kt
package com.walkud.self.mvp.base

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.widget.Toast

open class BaseActivity:AppCompatActivity(){

    fun showToast(msg :String){
        Toast.makeText(this,msg,Toast.LENGTH_SHORT).show()
    }

    fun forward(cls:Class<*>){
        val intent = Intent(this,cls)
        startActivity(intent)
    }
}